ArrayOperations/Arrays.cs

```diff
@@ -0,0 +1,324 @@
+      }
+
+      var output = new int[input.Length - 1];
+      for (var i = 0; i < output.Length; i++)
+      {
+        output[i] = i < index ? input[i] : input[i + 1];
+      }
+
+      return output;
+    }
+    
+    /// <summary>
+    /// Removes the first element from the array.
+    /// </summary>
+    /// <param name="input">Source array.</param>
+    /// <returns>A new array with it's size decreased by one and without the first element.</returns>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The source array has no elements.</exception>
+    public static int[] RemoveFirst(int[] input)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+
+      const int index = 0;
+      var output = RemoveAt(input, index);
+      return output;
+    }
+
+    /// <summary>
+    /// Removes the last element from the array.
+    /// </summary>
+    /// <param name="input">Source array.</param>
+    /// <returns>A new array with it's size decreased by one and without the last element.</returns>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The source has no elements.</exception>
+    public static int[] RemoveLast(int[] input)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+      
+      var output = RemoveAt(input, input.Length - 1);
+      return output;
+    }
+
+    /// <summary>
+    /// Prints the elements of an array separated by an empty space.
+    /// </summary>
+    /// <param name="input">Source array.</param>
+    /// <param name="message">Explanatory message displayed to the user.</param>
+    /// <param name="separator"></param>
+    /// <exception cref="ArgumentNullException">The source array is not initialized.</exception>
+    /// <exception cref="ArgumentException">The source has no elements.</exception>
+    /// <exception cref="ArgumentException">The message is null or empty.</exception>
+    public static void Print(int[] input, string message)
+    {
+      if (input == null)
+      {
+        throw new ArgumentNullException(nameof(input));
+      }
+
+      if (input.Length == 0)
+      {
+        throw new ArgumentException("Value cannot be an empty collection.", nameof(input));
+      }
+      
+      if (string.IsNullOrWhiteSpace(message))
+      {
+        throw new ArgumentException("Value cannot be null or empty.", nameof(message));
+      }
+
+      Console.WriteLine(message);
```

Comment:
Although it does shorten things I think having `message` here violates the "single responsibility principle" (more on that later) and so I'd keep...
```
Console.WriteLine(message);
Print(arr);
```
over trying to combine the two.

Reply:
I understand. Thank you.
Does the 'PromptString' method from the 'LoginSimulator' violate SRP?

Reply:
It's not always black&white.  I think PromptString is fine.